pkg/subscriptions/annotations_test.go

```diff
@@ -28,38 +28,53 @@ func TestIterate(t *testing.T) {
 			annotations: map[string]string{
 				"notifications.argoproj.io/subscribe.my-trigger.slack": "my-channel",
 			},
-			trigger:    "my-trigger",
-			service:    "slack",
-			recipients: []string{"my-channel"},
-			key:        "notifications.argoproj.io/subscribe.my-trigger.slack",
+			triggerList: []string{"my-trigger"},
+			service:     "slack",
+			recipients:  []string{"my-channel"},
+			key:         "notifications.argoproj.io/subscribe.my-trigger.slack",
 		},
 		{
 			annotations: map[string]string{
 				"notifications.argoproj.io/subscribe..slack": "my-channel",
 			},
-			trigger:    "",
-			service:    "slack",
-			recipients: []string{"my-channel"},
-			key:        "notifications.argoproj.io/subscribe..slack",
+			triggerList: []string{},
+			service:     "slack",
+			recipients:  []string{"my-channel"},
+			key:         "notifications.argoproj.io/subscribe..slack",
 		},
 		{
 			annotations: map[string]string{
 				"notifications.argoproj.io/subscribe.slack": "my-channel",
 			},
-			trigger:    "",
-			service:    "slack",
-			recipients: []string{"my-channel"},
-			key:        "notifications.argoproj.io/subscribe.slack",
+			triggerList: []string{},
+			service:     "slack",
+			recipients:  []string{"my-channel"},
+			key:         "notifications.argoproj.io/subscribe.slack",
+		},
+		{
+			annotations: map[string]string{
+				"notifications.argoproj.io/subscribe.my-trigger,my-trigger-2,my-trigger-3.slack": "my-channel",
```

Comment:
Annotation name cannot be longer than 63 characters: https://github.com/argoproj-labs/argocd-notifications/issues/255 . Multiple triggers most likely won't fit into 63 characters.

I would propose the following approach instead. Support annotation `notifications.argoproj.io/subscribe_yaml` that has YAML serialized information about all subscriptions

Example:

```yaml
annotations:
  notifications.argoproj.io/subscribe_yaml: |
    - trigger: my-trigger1
      destinations:
        - service: my-service1
          recipients:
            - recipient-1
            - recipient-2
    - trigger: my-trigger2
      destinations:
        - service: my-service2
          recipients:
            - recipient-3
            - recipient-4
```

The next step is to allow having subscript as a first-class field in target k8s resource:

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
spec:
  subscriptions:
    - trigger: my-trigger1
      destinations:
        - service: my-service1
          recipients:
            - recipient-1
            - recipient-2
    - trigger: my-trigger2
      destinations:
        - service: my-service2
          recipients:
            - recipient-3
            - recipient-4
```

Let me know what you think please

Reply:
@jessesuen I would very much appreciate your input as well

Reply:
Thanks for pointing out the character limit on annotation.
I like the approach but I want tweak a little bit on this..
instead of having a destination declared for every trigger.. can we have a destination for a set of triggers.. as this is the original problem I intended to solve for.. as rewriting destination for every trigger could be an overhead.
so can we have something like this:

```
  notifications.argoproj.io/subscribe_yaml: |
    - trigger: my-trigger1, my-trigger2, my-trigger3
      destinations:
        - service: my-service1
          recipients:
            - recipient-1
            - recipient-2
    - trigger: my-trigger4, my-trigger5, my-trigger6
      destinations:
        - service: my-service2
          recipients:
            - recipient-3
            - recipient-4
```

please let me know what you think on this approach?

Also is this comment `subscript as a first-class field in target k8s resource` to make argocd understand the subscriptions?

Reply:
> @jessesuen I would very much appreciate your input as well

@alexmt I recall there was a reason we chose to go with an annotation based approach vs. explicit spec. Do you recall why?